### DISCUSSION TITLE
Quarkus Camel Threading Model

### DISCUSSION THREAD

**Discussion by EricVincent3** [quarkusio/quarkus#34327] [Q&A]
I have a case where I need to reject my incoming request if my camel queue is maxed out, i tried to set my rejectedPolicy to Abort/Discard(by default its CallerRuns) but i don't see any rejected request... i've tried to set my max queue size to 100 and hit it with 200 thread using jmeter, but none of the request is rejected. i set my backend to return a response after 10s, so i'm assuming my first 100 request will fill the queue, and the other 100 will be rejected.. did i misunderstand something or what?
these is the properties i used to set the value.
![image](https://github.com/quarkusio/quarkus/assets/105971678/95a39581-932b-4d5e-9efd-a83f6c6b7580)

camel.threadpool.max-queue-size=100
camel.threadpool.rejected-policy=ABORT

thanks in advance

**Comment by gsmet** [comment]
@EricVincent3 I will ping @ppalaga and @zbendhiba but I think this question is probably better asked in Camel discussion channels.

**Comment by orpiske** [comment]
@EricVincent3 those are the settings for the (mostly) routing engine's thread pool and may or may not affect the component side of the equation. As in: in some cases the threads handling the component (lower level) part of the workload may be setup/taken/managed by a thread pool from the underlying component itself.

So, for a better answer, we would need to know which component you are using. Also, you are welcome to raise the question on our [user's mailing list](https://camel.apache.org/community/mailing-list/) or on [Zulip Chat](https://camel.zulipchat.com). 

We'll be happy to assist.
